hydroengine_service/dgds_functions.py

```diff
@@ -0,0 +1,427 @@
+DATASET_DIR = os.path.join(APP_DIR, 'datasets')
+with open(DATASET_DIR + '/dataset_visualization_parameters.json') as json_file:
+    DATASETS_VIS = json.load(json_file)
+
+logger = logging.getLogger(__name__)
+
+def get_dgds_source_vis_params(source, image_id=None):
+    """
+    Check source and/or image_id has default visualization parameters defined
+    :param source: String, source/location of Earth Engine Object
+    :param image_id: String, Earth Engine Image id
+    :return:
+    """
+    data_params = DATASETS_VIS.get(source, None)
+    if image_id and not data_params:
+        data_params = DATASETS_VIS.get(image_id, None)
+    assert data_params, f'{source} not in assets.'
+    return data_params
+
+
+def get_dgds_data(source,
+                  dataset=None,
+                  image_id=None,
+                  band=None,
+                  function=None,
+                  start_date=None,
+                  end_date=None,
+                  image_num_limit=None):
+    """
+
+    :param source: String, source/location of Earth Engine Object
+    :param dataset: String, name of Earth Engine Object dataset
+    :param image_id: String, Earth Engine Image id
+    :param band: String, band name to select from image
+    :param function: String, function to apply to image
+    :param start_date: String, start date to filter collection on
+    :param end_date: String, end date to filter collection on
+    :param image_num_limit: String, limit of objects to return
+    :return: Dictionary
+    """
+    # Get list of objects with imageId and date for collection
+    data_params = get_dgds_source_vis_params(source, image_id)
+    info = get_image_collection_info(source, start_date, end_date, image_num_limit)
+    if not info:
+        return
+
+    # get most recent to return url
+    returned_url_id = info[-1]["imageId"]
+    if image_id:
+        returned_url_id = image_id
+
+    if data_params.get('function', None) and not function:
+        function = data_params['function']
+        if isinstance(function, list):
+            function = function[0]
+        else:
+            function = function.get(band, None)
+
+    image_info = _get_wms_url(returned_url_id, type=data_params['type'], band=band, function=function)
+    image_info['dataset'] = dataset
+    image_info['band'] = band
+    image_info['imageTimeseries'] = info
+
+    return image_info
+
+def visualize_gebco(source, band):
+    """
+    Specialized function to visualize GEBCO data
+    :param source: String, Google Earth Engine image id
+    :param band: String, band of image to visualize
+    :return: Dictionary
+    """
+    data_params = DATASETS_VIS[source]
+    image = ee.Image(source)
+
+    gebco = image.select(data_params['bandNames'][band])
+    # Angle for hillshade (keep at 315 for good perception)
+    azimuth = 315
+    # Lower is longer shadows
+    zenith = 30
+
+    bathy_only = data_params.get('bathy_only', False)
```

Comment:
Maybe nice to group vis parameters together that you want to explain separately.